cwms_radar_api/src/main/java/cwms/radar/data/dto/Location.java

```diff
@@ -1,24 +1,387 @@
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(getName(), getLatitude(), getLongitude(), active(), getPublicName(), getLongName(), getDescription(), getTimezoneId(), getLocationType(), getLocationKind(), getNation(), getStateInitial(), getCountyName(), getHorizontalDatum(), getPublishedLongitude(), getPublishedLatitude(), getVerticalDatum(), getElevation(), getMapLabel(), getBoundingOfficeId(), getOfficeId());
+    }
+
+    @JsonPOJOBuilder
+    public static class Builder
+    {
+        private String name;
+        private double latitude;
+        private double longitude;
+        private final String officeId;
+        private boolean active = true;
+        private String publicName;
+        private String longName;
+        private String description;
+        private String timezoneId;
+        private String locationType;
+        private String locationKind;
+        private Nation nation;
+        private String stateInitial;
+        private String countyName;
+        private String nearestCity;
+        private String horizontalDatum;
+        private Double publishedLongitude;
+        private Double publishedLatitude;
+        private String verticalDatum;
+        private Double elevation;
+        private String mapLabel;
+        private String boundingOfficeId;
+        private static final String MISSING_NAME_ERROR_MSG = "Location name is required";
+
+        @JsonCreator
+        public Builder(@JsonProperty(value = "name", required = true) String name, @JsonProperty(value = "locationKind", required = true) String locationKind,
+                       @JsonProperty (value = "timezoneId", required = true) ZoneId timeZoneId, @JsonProperty(value = "latitude", required = true) double latitude,
```

Comment:
In the Feature/locations xml branch I updated this so the `required = true` is removed, that way it allows for updating a single parameter of a location, otherwise an error would be thrown. Because of that, now if there is a required field missing, that gets checked on storage, not the create of the DTO object.